core/server/master/src/main/java/alluxio/master/meta/ServerConfigurationChecker.java

```diff
@@ -13,26 +13,34 @@
 
 import alluxio.wire.ConfigProperty;
 
-import java.util.Collections;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
 /**
  * This class is responsible for checking server-side configuration.
  */
 public class ServerConfigurationChecker {
   /** Map from a node id to its configuration. */
   private Map<Long, List<ConfigProperty>> mConfMap;
-  /** Map from a lost node id to its configuration. */
-  private Map<Long, List<ConfigProperty>> mLostConfMap;
+  /** Set that contains the ids of lost nodes. */
+  private Set<Long> mLostNodes;
 
   /**
    * Constructs a new {@link ServerConfigurationChecker}.
    */
   public ServerConfigurationChecker() {
+    init();
+  }
+
+  /**
+   * Initializes the default {@link ServerConfigurationChecker}.
+   */
+  public synchronized void init() {
     mConfMap = new HashMap<>();
-    mLostConfMap = new HashMap<>();
+    mLostNodes = new HashSet<>();
```

Comment:
We can make `mConfMap` and `mLostNodes` final if we instead use `mConfMap.clear()` and `mLostNodes.clear()`. This also makes it easier for the garbage collector to collect the old configurations.